File: 9-25数字拼接排序164.cpp
#include<vector>
#include<string>
        
class Solution {
    public:
        string crackPassword(vector<int>& password) {
            for (int i=0;i<password.size();i++)
            {
                for(int j=0;j<password.size()-1;j++)
                {
                    if not(compare(password[j],password[j+1]))
                    {swap(password[j],password[j+1]);}
                }
            }
            string res="";
            for(int i=0;i<password.size();++i)
            {
                res+=to_string(password[i]);
            }
            return res;
            
        }
        bool compare(int a,int b)
        {
            if (to_string(a)+to_string(b)<to_string(b)+to_string(a))
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    };

File: 9-25数字拼接排序164.py
class Solution:
    def crackPassword(self, password: List[int]) -> str:
        strlist=[str(i) for i in password]
        def compare(s1,s2):
            if s1+s2>s2+s1:
                return 1
            elif s1+s2<s2+s1:
                return -1
            else:
                return 0
        sorted_list=sorted(strlist,key=functools.cmp_to_key(compare))
        # for i in range(len(strlist)):
        #     for j in range(0,len(strlist)-1):
        #         if compare(strlist[j],strlist[j+1])>0:
        #             strlist[j],strlist[j+1]=strlist[j+1],strlist[j]
        #         else:
        #             continue
        return ''.join(sorted_list)
        
